pkg/bloomutils/ring.go

```diff
@@ -14,20 +14,47 @@ import (
 	v1 "github.com/grafana/loki/pkg/storage/bloom/v1"
 )
 
-type InstanceWithTokenRange struct {
-	Instance           ring.InstanceDesc
-	MinToken, MaxToken uint32
+type integer interface {
```

Comment:
I think you can use `constrainsts.Integer` instead:
https://pkg.go.dev/golang.org/x/exp/constraints#Integer